syntax = "proto3";

message Person {
    int32 id = 1;
    string first_name = 2;
    string last_name = 3;
    string company_name = 4;
}

message Location {
    int32 id = 1;
    int32 person_id = 2;
    repeated float coordinates = 3;
    string creation_time = 4;
}

message Connection {
    Location location = 1;
    Person person = 2;
}

message ConnectionRequest {
    Location location = 1;
    string start_date = 2;
    string end_date = 3;
    int32 meters = 4;
}

service ConnectionTracker {
    rpc Get(ConnectionRequest) returns (stream Connection);
}